ui/src/userInput/AzExtUserInput.ts

```diff
@@ -26,13 +27,19 @@ export class AzExtUserInput implements types.IAzureUserInput {
 
     public async showQuickPick<TPick extends types.IAzureQuickPickItem<unknown>>(picks: TPick[] | Promise<TPick[]>, options: types.IAzureQuickPickOptions): Promise<TPick | TPick[]> {
         addStepTelemetry(this._context, options.stepName, 'quickPick', options.placeHolder);
+        if (this._context.ui.wizard?.cancellationToken.isCancellationRequested) {
+            throw new UserCancelledError();
+        }
         const result = await showQuickPick(this._context, picks, options);
         this._onDidFinishPromptEmitter.fire({ value: result });
         return result;
     }
 
     public async showInputBox(options: types.AzExtInputBoxOptions): Promise<string> {
         addStepTelemetry(this._context, options.stepName, 'inputBox', options.prompt);
+        if (this._context.ui.wizard?.cancellationToken.isCancellationRequested) {
```

Comment:
We need this logic for `showOpenDialog` and `showWarningMessage` as well - they can be used in a wizard step, too.